test/proposals.unit.tests.js

```diff
@@ -0,0 +1,329 @@
+			await preserveBalancesOnTransferToken.transferOwnership(proposalsContract.address);
+			await bridgeTestable.transferOwnership(proposalsContract.address);			
+		});
+
+		it('Should assert.equal epochLength as 0',async() => {
+			var EL1 = await bridgeTestable.epochLength();
+			assert.equal(EL1.toNumber(), 0);			
+		});
+
+		it('Should create proposal for setEpochLength',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data);
+		});
+
+		it('Should not create proposal for setEpochLength by outsider',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data, {from:outsider}).should.be.rejectedWith('revert');
+		});		
+
+		it('Should assert equal epochLength as 0 after proposal creation',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			var EL1 = await bridgeTestable.epochLength();
+			assert.equal(EL1.toNumber(), 0);
+		});		
+
+		it('Should vote true for EpochLength',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			await proposalsContract.veto(0,{from:u1});
+		});
+
+		it('Should not vote if no proposal',async() => {
+			await proposalsContract.veto(0, {from:u1}).should.be.rejectedWith('revert');
+		});
+
+		it('Should not vote if wrong id',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			await proposalsContract.veto(1, {from:u1}).should.be.rejectedWith('revert');
+		});
+
+		it('Should not veto if voter have no tokens',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			await proposalsContract.veto(0, {from:outsider}).should.be.rejectedWith('revert');
+		});		
+
+		it('Should create two differnt proposals + can veto',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(500);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			var data = await bridgeTestable.contract.setEpochLength.getData(700);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			await proposalsContract.veto(0, {from:u1});
+			await proposalsContract.veto(1, {from:u1});
+		});
+
+		it('Should not change epochLength if vetoed',async() => {
+			var data = await bridgeTestable.contract.setEpochLength.getData(700);
+			await proposalsContract.propose(bridgeTestable.address, data);
+			await proposalsContract.veto(0, {from:u1});
+			await proposalsContract.veto(0, {from:u2});
+			await time.increase(time.duration.days(14));
+			var EL2 = await bridgeTestable.epochLength();
```

Comment:
I suggest we test here that proposal is finalized, otherwise it is not obvious that `epochLength` didn't change because it was vetoed (it won't change as well if proposal is not finalized).